scripts/bootloader/keygen.py

```diff
@@ -26,11 +42,13 @@
                         help='Output to specified file instead of stdout.')
     parser.add_argument('--in', '-in', '-i', required=False, dest='infile',
                         type=argparse.FileType('rb'),
-                        help='Read private key from specified PEM file instead of generating it.')
+                        help='Read private key from specified PEM file instead '
+                             'of generating it.')
 
     args = parser.parse_args()
 
-    sk = SigningKey.from_pem(args.infile.read()) if args.infile else SigningKey.generate(curve=NIST256p)
+    sk = (SigningKey.from_pem(args.infile.read())
```

Comment:
just wondering.
If user provides an in-file, and the digest of that key contains `0xFFFF` then I assume we still have a problem.

I know we can't change the key, but we could print a big warning, informing the user of a potential issue with his key.

Should we do that ?

Reply:
This is checked at a later stage, in another script IIRC.

Reply:
Reading NCSDK-10060 it does state that build system will fail, but not at which stage, CMake, compilation, linking, post-processing, and how.

If it fails at CMake configure stage, then that's good, if it fails later, I think an earlier warning would be nice to the user.

Reply:
This PR only addresses the scenario when the user does not provide a key. If a key is provided who's hash contains 0xFFFF it will be catched in the provision step (now described in the task you link). It will also fail runtime.